sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -12,18 +15,75 @@ namespace core
 namespace http
 {
 
-class http_request
+namespace http_method
+{
+enum HttpMethod
 {
+  GET,
+  HEAD,
+  POST,
+  PUT,
+  DELETE,
+  PATCH,
+};
+}
 
+class Header
+{
 private:
-  /* data */
+  std::string name;
+  std::string value;
+
 public:
-  http_request();
-  ~http_request();
+  Header(std::string name, std::string value)
+  {
+    this->name = name;
+    this->value = value;
+  }
+  ~Header() {}
+  std::string getName();
+  std::string getValue();
 };
 
-http_request::http_request() {}
-http_request::~http_request() {}
+class Request
+{
+
+private:
+  http_method::HttpMethod method;
+  std::string url;
+  std::vector<Header> headers;
+  std::string body;
+  size_t query_start; // 0 = no query in url, > 0 = query start position in url '?'
+
+  inline size_t get_query_start(std::string someUrl)
```

Comment:
Member function defined in class body is inlined by default. the `inline` keyword is superfluous

Reply:
I think `someUrl` should be passed by const reference, because we just want to read it, we don't need to make a copy.

Reply:
Plus, this function doesn't access any member variables or functions, we should make it static.